x/valset-pref/validator_set.go

```diff
@@ -291,8 +291,8 @@ func (k Keeper) UndelegateFromRebalancedValidatorSet(ctx sdk.Context, delegatorA
 			// Use the min between our undelegation amount calculated via iterations of undelegating
 			// and the amount actually delegated to the validator. This is done to prevent an error
 			// in the event some rounding issue increases our calculated undelegation amount.
-			delegation, found := k.stakingKeeper.GetDelegation(ctx, delegator, val.ValAddr)
-			if !found {
+			delegation, err := k.stakingKeeper.GetDelegation(ctx, delegator, val.ValAddr)
+			if err != nil {
 				return fmt.Errorf("No delegation found for delegator %s to validator %s\n", delegator, val.ValAddr)
 			}
```

Comment:
Yeah agreed with this, for simplicity I am just bubbling up the error, because the sdk returns an error now instead of false if the value is nil (https://github.com/osmosis-labs/cosmos-sdk/blob/1a5662f2a4586735a64388386eaf23a62dfddad9/x/staking/keeper/delegation.go#L30-L32)

Previously we still returned the delegation even if the value was nil (https://github.com/cosmos/cosmos-sdk/blob/7009a2e0cdb93b286199bbaa22e8880aeec83c8c/x/staking/keeper/delegation.go#L15-L27)

Change here  https://github.com/osmosis-labs/osmosis/pull/8274/commits/c780c738545051db2fa22c25a410f416ed143068